### PR TITLE
feat(api): cut snippets at a word boundary and say so

### PR DESCRIPTION
**Pull request by jan-kubica**
Add an explicit `truncated` field to public decision-row headnote previews.
Cut over-limit text at Unicode word boundaries under the existing row limit.
Bind the shared type to a closed response schema and preserve the ellipsis in the web row.
Advance the public REST contract revision.

## Summary by CodeRabbit

* **New Features**
  * Decision headnotes now provide bounded previews with explicit truncation metadata.
  * Truncated headnotes retain an ellipsis and end at readable word boundaries.
  * API responses use a consistent format for present and absent headnotes.
* **Bug Fixes**
  * Improved validation and handling of headnote length, truncation, and Unicode text.
* **Tests**
  * Added coverage for valid, invalid, absent, and truncated headnote previews, including boundary cases.

### CODE REVIEW

File: apps/api/src/lib/case-law/decision-text.ts [hunk 1]
```diff
@@ -29,7 +30,7 @@ const DECISION_TEXT_FIELD_KEY_SET = new Set<string>(DECISION_TEXT_FIELD_KEYS);
 const isDecisionTextFieldKey = (key: string): key is DecisionTextFieldKey =>
   DECISION_TEXT_FIELD_KEY_SET.has(key);
 
-export const absentTextField = (reason: TextAbsenceReason): TextField => ({
+export const absentTextField = (reason: TextAbsenceReason) => ({
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/lib/case-law/decision-text.ts:33]: Confirmed on the current head: `bun --filter @stll/api typecheck` exits 0. `TEXT_FIELD_TYPE.ABSENT` is a non-widening `as const` member, so this helper infers the literal absent branch. Keeping that narrow branch is intentional because `readDecisionHeadnote` returns the absent branch of `DecisionHeadnotePreview`; annotating the helper as the full `TextField` union would discard that precision. No code change.

CC on behalf of jan-kubica

File: apps/api/src/lib/case-law/decision-headnote.ts [hunk 2]
```diff
@@ -1,37 +1,42 @@
...
+ * still distinguishes it from an absent publisher summary.
  */
-export const normalizeDecisionHeadnote = (raw: unknown): string | null => {
+export const truncateDecisionHeadnote = (text: string) => {
+  const max = LIMITS.caseLawHeadnoteMaxChars;
+  if (text.length <= max) {
+    return { text, truncated: false };
+  }
+  let cut = 0;
+  const boundaryProbe = text.slice(0, max + 1);
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/lib/case-law/decision-headnote.ts]: Fixed in `f43c68d1f3`. Segmentation now iterates the original text and stops before the next complete segment exceeds the content budget; it no longer creates an artificial boundary by pre-slicing. A regression covers a limit crossing inside `🇨🇿`, and the focused property suite plus API typecheck pass.

CC on behalf of jan-kubica

File: apps/api/src/lib/case-law/decision-headnote.ts [hunk 3]
```diff
@@ -1,37 +1,42 @@
...
+  for (const { index, segment } of WORD_SEGMENTER.segment(boundaryProbe)) {
+    const segmentEnd = index + segment.length;
+    if (segmentEnd > max) {
+      break;
+    }
+    cut = segmentEnd;
+  }
+  return { text: text.slice(0, cut).trimEnd(), truncated: true };
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/api/src/lib/case-law/decision-headnote.ts]: Fixed in `f43c68d1f3`. The API now retains the terminal ellipsis inside `headnote.text` while also returning `truncated`; the web renders that text verbatim. Old and new bundle/API combinations therefore keep the visible marker, and it remains inside the existing row-size budget. Focused tests and both API/web typechecks pass.

CC on behalf of jan-kubica
